alert/tests.py

```diff
@@ -532,21 +534,341 @@ def test_no_permission(self):
+        self.assertEqual(response.data['last_name'], 'newname')
+
+    def test_update_settings_change_username(self):
+        response = self.client.put('/api/settings/',
+                                   json.dumps({'username': 'newusername', 'first_name': '', 'last_name': '',
+                                               'profile': {'email': 'example@email.com', 'phone': '3131234567'}}),
+                                   content_type='application/json')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['profile']['email'], 'example@email.com')
+        self.assertEqual(response.data['profile']['phone'], '+13131234567')
+        self.assertEqual(response.data['username'], 'newusername')
+        self.assertEqual(response.data['first_name'], '')
+        self.assertEqual(response.data['last_name'], '')
+        response = self.client.get('/api/settings/')
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(response.data['profile']['email'], 'example@email.com')
+        self.assertEqual(response.data['profile']['phone'], '+13131234567')
+        self.assertEqual(response.data['username'], 'newusername')
+        self.assertEqual(response.data['first_name'], '')
+        self.assertEqual(response.data['last_name'], '')
+
+    def test_add_fields(self):
+        response = self.client.put('/api/settings/',
+                                   json.dumps({'username': 'jacob', 'first_name': '', 'last_name': '',
+                                               'middle_name': 'm',
+                                               'profile': {'email': 'example@email.com', 'phone': '3131234567',
+                                                           'favorite_color': 'blue'}}),
+                                   content_type='application/json')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['profile']['email'], 'example@email.com')
+        self.assertEqual(response.data['profile']['phone'], '+13131234567')
+        self.assertFalse('favorite_color' in response.data['profile'])
+        self.assertEqual(response.data['username'], 'jacob')
+        self.assertEqual(response.data['first_name'], '')
+        self.assertEqual(response.data['last_name'], '')
+        self.assertFalse('middle_name' in response.data)
+        response = self.client.get('/api/settings/')
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(response.data['profile']['email'], 'example@email.com')
+        self.assertEqual(response.data['profile']['phone'], '+13131234567')
+        self.assertFalse('favorite_color' in response.data['profile'])
+        self.assertEqual(response.data['username'], 'jacob')
+        self.assertEqual(response.data['first_name'], '')
+        self.assertEqual(response.data['last_name'], '')
+        self.assertFalse('middle_name' in response.data)
+
+    def test_ignore_fields_email_update(self):
```

Comment:
why is this an `ignore_fields` test? looks like it's correctly updating emails

Reply:
The bulk of it is the request:

```
response = self.client.put('/api/settings/',
                                   json.dumps({'profile': {'email': 'example2@email.com'}}),
                                   content_type='application/json')
```

Which tests that you can update fields without including other fields and you won't have any issues or unwanted updates.